flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -119,6 +154,66 @@ class DataStreamOverAggregate(
 
   }
 
+  def createTimeBoundedProcessingTimeOverWindow(
+    inputDS: DataStream[Row]): DataStream[Row] = {
+
+    val overWindow: Group = logicWindow.groups.get(0)
+    val partitionKeys: Array[Int] = overWindow.keys.toArray
+    val namedAggregates: Seq[CalcitePair[AggregateCall, String]] = generateNamedAggregates
+
+    val index = overWindow.lowerBound.getOffset.asInstanceOf[RexInputRef].getIndex
+    val count = input.getRowType().getFieldCount()
+    val lowerboundIndex = index - count
+    var time_boundary = 0L
+     
+    logicWindow.constants.get(lowerboundIndex).getValue2 match {
+      case _: java.math.BigDecimal => time_boundary = logicWindow.constants.get(lowerboundIndex)
+         .getValue2.asInstanceOf[java.math.BigDecimal].longValue()
+      case _ => throw new TableException("OVER Window boundaries must be numeric")
+    }
+
+    val (aggFields, aggregates) =  AggregateUtil.transformToAggregateFunctions(
+         namedAggregates.map(_.getKey),inputType, needRetraction = false)
+         
+         
+    // As we it is not possible to operate neither on sliding count neither
+    // on sliding time we need to manage the eviction of the events that
+    // expire ourselves based on the proctime (system time). Therefore the
+    // current system time is assign as the timestamp of the event to be
+    // recognize by the evictor
+
+    val inputDataStreamTimed = inputDS
+      .assignTimestampsAndWatermarks(new ProcTimeTimestampExtractor())
+
+    // get the output types
+    val rowTypeInfo = FlinkTypeFactory.toInternalRowTypeInfo(getRowType).asInstanceOf[RowTypeInfo]
+
+    val result: DataStream[Row] =
+      if (partitionKeys.nonEmpty) {
+        inputDataStreamTimed.keyBy(partitionKeys:_*)
+          .window(GlobalWindows.create())
+          .trigger(CountTrigger.of(1))
+          .evictor(TimeEvictor.of(Time.milliseconds(time_boundary))) 
+          .apply(new DataStreamProcTimeAggregateWindowFunction[GlobalWindow]
+               (aggregates,aggFields,inputType.getFieldCount))
+          .returns(rowTypeInfo)
+          .name(aggOpName)
+          .asInstanceOf[DataStream[Row]]
+        
+      } else {
+          inputDataStreamTimed.windowAll(GlobalWindows.create()).trigger(CountTrigger.of(1))
+            .evictor(TimeEvictor.of(Time.milliseconds(time_boundary)))        
+            .apply(new DataStreamProcTimeAggregateGlobalWindowFunction[GlobalWindow](
```

Comment:
I think a `ProcessFunction` would be more efficient for this use case. With a `WindowFunction` and a `GlobalWindow`, we have to aggregate all rows for each emitted row. With a `ProcessFunction` we can just add the new row and retract those rows which are not included anymore.

Moreover, we have to change the logic to a `ProcessFunction` anyway when we extend this to support `FOLLOWING`